src/acl_hal_mmd.cpp

```diff
@@ -2854,3 +2859,11 @@ void acl_hal_mmd_simulation_streaming_kernel_done(
       .mmd_dispatch->aocl_mmd_simulation_streaming_kernel_done(
           device_info[physical_device_id].handle, kernel_name, finish_counter);
 }
+
+void acl_hal_mmd_simulation_set_kernel_cra_address_map(
+    unsigned int physical_device_id,
+    const std::vector<uintptr_t> kernel_csr_address_map) {
```

Comment:
One more nitpick, I think it's better to pass kernel_csr_address_map by reference.